internal/cmd/gocdk/init.go

```diff
@@ -15,11 +15,101 @@
 package main
 
 import (
+	"bytes"
 	"context"
+	"flag"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
 
+	"gocloud.dev/internal/cmd/gocdk/internal/templates"
 	"golang.org/x/xerrors"
 )
 
 func init_(ctx context.Context, pctx *processContext, args []string) error {
-	return xerrors.New("not implemented")
+	f := newFlagSet(pctx, "init")
+	var modpath string
+	f.StringVar(&modpath, "module-path", "", "the module path for your project's go.mod file")
+	f.StringVar(&modpath, "m", "", "alias for --module-path")
+
+	if err := f.Parse(args); xerrors.Is(err, flag.ErrHelp) {
+		return nil
+	} else if err != nil {
+		return usagef("gocdk init: %w", err)
+	}
+
+	if f.NArg() != 1 {
+		return usagef("gocdk init [options] PATH")
```

Comment:
Did we consider dropping the `PATH` argument and having `gocdk init` initialize the current directory? I.e., you'd do:

```
$ mkdir myproject
$ cd myproject
$ gocdk init
```

Reply:
Yes: #1918 tracks this sort of usage.

Reply:
I meant instead of this, not in addition...

Reply:
We talked about that, but then there was a fair amount of back and forth about if it needed to be in an empty directory and then what constituted "empty", and the decision was made for simplicity sake to create a new directory, at least at first.

Reply:
OK.